streamer/src/nonblocking/quic.rs

```diff
@@ -1152,7 +1152,16 @@ pub mod test {
         for _ in 0..num_bytes {
             s1.write_all(&[0u8]).await.unwrap();
         }
-        s1.finish().await.unwrap();
+        let result = s1.finish().await;
+
+        // The call to finish, after sending multiple writes on quic in the
+        // test, is to make sure that we Stop writing any bytes on the connect.
+        // So when the test call finishes after sending multiple writes on quic,
+        // a Stopped from peer should be fine. We expect a result of Ok or
+        // Stopped from peer.
+        if result.is_err() && !matches!(result, Err(quinn::WriteError::Stopped(_))) {
+            panic!("Failed to finish writes on quic connection");
+        }
```

Comment:
```suggestion
        match s1.finish().await {
            Ok() => (),
            // Call to finish is to make sure we Stop writing more bytes.
            // So when finish is called after sending multiple writes on quic,
            // Stopped error from peer is fine.
            quinn::WriteError::Stopped(_) => (),
            _ => panic!("Failed to finish writes on quic connection");
        }
```

Reply:
+1

Also, I would include the error in the error message:

```rust
        match s1.finish().await {
            // Call to finish is to make sure we Stop writing more bytes.
            // So when finish is called after sending multiple writes on quic,
            // Stopped error from peer is fine.
            Ok(()) | Err(quinn::WriteError::Stopped(_)) => (),
            Err(err) => panic!("Failed to finish writes on quic connection: {err}");
}
```